### PR TITLE
Fix line break preservation in editor parser

### PR DESCRIPTION
**Pull request by thesocialdev**
## Summary
- Fix editor parser bug where line breaks are removed when saving reports
- Preserve paragraph structure with proper multi-paragraph content handling
- Implement Guard Clause pattern instead of ternaries for better readability
- Add comprehensive test coverage for line break preservation and source annotations
- Ensure backward compatibility with existing content

## Test plan
- [x] Multi-paragraph content preserves line breaks when saved
- [x] HTML output correctly renders paragraphs with proper spacing
- [x] Source annotations work correctly with line breaks
- [x] Empty paragraphs are handled gracefully
- [x] Backward compatibility maintained for existing content
- [x] All existing tests continue to pass
- [x] New comprehensive test suite covers edge cases

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: lib/editor-parser.ts [hunk 1]
```diff
@@ -84,10 +127,19 @@ export class EditorParser {
...
         }
 
-        return `<div><p>${content}</p></div>`;
+        return this.buildSingleHtmlContent(String(content));
+    }
+
+    private buildSingleHtmlContent(contentStr: string): string {
+        if (this.isMultiParagraphContent(contentStr)) {
+            const htmlContent = this.convertLineBreaksToHtml(contentStr);
+            return `<div><p>${htmlContent}</p></div>`;
```
[the remark this answers is not in the document]
Comment (thesocialdev) [reply lib/editor-parser.ts:139]: That's a great question. Yes, but not for empty lines.
